pkg/checksdb/check.go

```diff
@@ -216,7 +217,7 @@ func (check *Check) SetResultFailed(reason string) {
 	}
 
 	check.Result = CheckResultFailed
-	check.FailureReason = reason
+	check.SkipReason = reason
```

Comment:
If the check failed it means that it has not been skipped, so the `SkipReason` should be empty. The reason for the failure is that there are non-compliant objects that can be found in `Details`.

Reply:
fixed. Removed SetResultFailed function as it is not used (setResult already handles the failed case)